### PR TITLE
Doc: Restructure Contributing guide

### PR DESCRIPTION
**Pull request by sgillespie**
# Description

Restructure Contributing document. The purpose is to make it easier to add more content.

[Rendered](https://github.com/sgillespie/cardano-db-sync/blob/doc/contributing/CONTRIBUTING.rst)

# Checklist

- [x] Commit sequence broadly makes sense
- [x] Commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] Any changes are noted in the [changelog](https://github.com/IntersectMBO/cardano-db-sync/blob/master/cardano-db-sync/CHANGELOG.md)
- [ ] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) on version 0.10.1.0 (which can be run with `scripts/fourmolize.sh`)
- [x] Self-reviewed the diff

# Migrations

- [ ] The pr causes a [breaking change](https://github.com/IntersectMBO/cardano-db-sync/blob/master/doc/migrations.md) of type a,b or c
- [ ] If there is a breaking change, the pr includes a database migration and/or a fix process for old values, so that upgrade is possible
- [ ] Resyncing and running the migrations provided will result in the same database semantically

If there is a breaking change, especially a big one, please add a justification here. Please elaborate
more what the migration achieves, what it cannot achieve or why a migration is not possible.

### CODE REVIEW

**Comment by kderme** [comment]
Seems good. Could there be some words about using nix shell/develop and haskell-language-server?

**Comment by sgillespie** [comment]
> Seems good. Could there be some words about using nix shell/develop and haskell-language-server?

I've added a bit in there, let me know if you want me to expand or change it in any way

**Review by Cmdv — APPROVED** [review]
Lovely 👌

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -38,32 +149,94 @@ project.  The most important thing to note is that we pin the ``index-state`` of
...
-getting them from Hackage.  The differences are that it has an independent
-``index-state``, and that there is a different Nix command you need to run
-afterwards: ``nix flake lock --update-input CHaP``.
+Update the Nix equivalent of the newer ``index-state``::
+
+  nix flake lock --update-input CHaP
+
+If there are updated configuration files, be sure to also update ``iohk-nix``::
+
+  nix flake lock --update-input nix flake lock --update-input hackageNix
```
Comment (kderme) [inline CONTRIBUTING.rst:212]:
```suggestion
  nix flake lock --update-input hackageNix
```

**Review by kderme — CHANGES_REQUESTED** [review]

**Review by kderme — APPROVED** [review]

**Pull request merged**